tensorflow/compiler/mlir/tools/kernel_gen/kernel_creator.cc

```diff
@@ -403,6 +403,21 @@ Status LowerKernelBodiesToLowLevelIr(mlir::ModuleOp module) {
       "Neither TENSORFLOW_USE_ROCM nor GOOGLE_CUDA are defined."
       " Did you specify either --config=rocm or --config=cuda ?");
 #endif
+
+#if TENSORFLOW_USE_ROCM
+  auto gpu_modules = module.getOps<::mlir::gpu::GPUModuleOp>();
+  for (::mlir::gpu::GPUModuleOp gpu_module : gpu_modules) {
+    gpu_module.walk([&](mlir::gpu::GPUFuncOp gpu_kernel) {
+      if (gpu_kernel->getAttr(mlir::gpu::GPUDialect::getKernelFuncAttrName())) {
+        gpu_kernel->setAttr(
+            "rocdl.max_flat_work_group_size",
```

Comment:
CUDA has similar concepts. I think it would be desirable to add a max work_group_size at the GPU dialect level and then propagate this to the AMD specific information when going to LLVM.

Reply:
/cc @whchung @krzysz00 ... see comment from @sherhut above

Reply:
@sherhut Yeah, that might be a good idea. I can't quite follow up on it right now, but I agree that's a good path going forward

Reply:
@sherhut can we take this PR for now, as it is needed to fix a lot of unit test failures on ROCm.

Making the change to the GPU Dialect is non-trival and needs to be done independent of this PR

Reply:
Sure, we should not block this, especially as the llvm side change has landed. I do hope the cleanup happens eventually.

Can you address the small change above?

Reply:
@sherhut done.

Reply:
@sherhut gentle ping...please approve

Reply:
@sherhut gentle ping...please approve